src/Jackalope/Node.php

```diff
@@ -773,9 +785,10 @@ public function getPropertiesValues($nameFilter = null, $dereference = true)
         $this->checkState();
 
         // OPTIMIZE: do not create properties in constructor, go over array here
-        $names = self::filterNames($nameFilter, array_keys($this->properties));
+        $names = self::filterNames($nameFilter, array_keys($this->propertyValues));
         $result = array();
         foreach ($names as $name) {
+            $this->loadProperty($name);
```

Comment:
sorry i was unclear. we added this method (among other reasons) to provide ourselves a way to *in the future* provide access to the values without needing to instantiate properties.

however, its probably a bit tricky, depending on the type (reference, date, ...) we would need to convert a string value so it could get really quite hairy. we could check if we have the property and if not, create it unless its a string/int/float property in which case we can just use the value as is, wdyt?